File: Js/ShackerSort.js
const ShackerSort = (l) => {
  let sortedL = [...l];
  let i = 0;
  while (l.length >= i) {
    let [left, right] = [i, l.length - i];
    for (let j = left; j < right; j++) {
      if (sortedL[j - 1] > sortedL[j]) {
        [sortedL[j - 1], sortedL[j]] = [sortedL[j], sortedL[j - 1]];
      }
    }
    right--;
    for (let j = right - 1; j > left; j--) {
      if (sortedL[j - 1] > sortedL[j]) {
        [sortedL[j - 1], sortedL[j]] = [sortedL[j], sortedL[j - 1]];
      }
    }
    i++;
  }
  return sortedL;
};

const l = [2, 5, 1, 7, 3, 9, 43, 12, 13, 13, 54, 23, 11];
console.log(ShackerSort(l));

File: Python/ShackerSort.py
#ShackerSort
def ShackerSort(l):
    left=0
    right=len(l)
    while right>left:
        
        #Go from right side
        for a in range(left,right-1):
            if l[a]>l[a+1]:
                l[a],l[a+1]=l[a+1],l[a]
        right-=1
        
        #Go from left side
        for a in range(right,left,-1):
            if  l[a-1]>l[a]:
                l[a],l[a-1]=l[a-1],l[a]
        left+=1
    return l

#Create our List
ourList=[1,4,2,7,4,8,1,2,4,94,12]
print(ShackerSort(ourList))
